unit/java_bytecode/java_bytecode_parse_generics/parse_signature_descriptor_mismatch.cpp

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing generic classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <istream>
+#include <memory>
+
+#include <util/config.h>
+#include <util/language.h>
+#include <util/message.h>
+#include <java_bytecode/java_bytecode_language.h>
+#include <iostream>
+#include <testing-utils/load_java_class.h>
+
+SCENARIO(
+  "java_bytecode_parse_signature_descriptor_mismatch",
+  "[core][java_bytecode][java_bytecode_parse_generics]")
+{
+  const symbol_tablet &new_symbol_table=
+    load_java_class(
+      "Outer",
+      "./java_bytecode/java_bytecode_parse_generics");
+
+  const std::string class_prefix="java::Outer";
+  REQUIRE(new_symbol_table.has_symbol(class_prefix));
+
+  const std::string inner_prefix=class_prefix+"$Inner";
+  THEN("There is a (complete) symbol for the inner class Inner")
+  {
+    REQUIRE(new_symbol_table.has_symbol(inner_prefix));
+    new_symbol_table.show(std::cout);
+
+    const symbolt &inner_symbol=new_symbol_table.lookup_ref
```

Comment:
Nit: Either break the line after the `=` (my personal favourite) or after the `(`